client/src/main/java/io/atomix/copycat/client/session/ClientSession.java

```diff
@@ -580,12 +589,30 @@ private ClientSession resetConnection() {
    */
   private ClientSession resetMembers() {
     if (connectMembers.isEmpty() || connectMembers.size() < members.size() - 1) {
-      connectMembers = connectionStrategy.getConnections(leader, new ArrayList<>(members));
+      connectMembers = new ArrayList<>(members);
     }
     return this;
   }
 
   /**
+   * Selects the next member to which to send a request.
+   */
+  private Address selectMember() {
+    // Remove the next member that meets the strategy from the members list.
+    Iterator<Address> iterator = connectMembers.iterator();
+    while (iterator.hasNext()) {
+      Address member = iterator.next();
+      if (connectionStrategy.connect(member, leader)) {
+        iterator.remove();
```

Comment:
I feel we should consider the situation where a client can start up before the active members do. By discarding members that have been tried without success we'll eventually exhaust the list of servers and when that happens the client gives up. It would be good to make this configurable so that a client can choose to retry until a valid session can be established.